datasquare.go

```diff
@@ -163,6 +166,9 @@ func (ds *dataSquare) setColSlice(x uint, y uint, newCol [][]byte) error {
 			return errors.New("invalid chunk size")
 		}
 	}
+	if x+uint(len(newCol)) > ds.width {
+		return fmt.Errorf("cannot set col slice at (%d, %d) of length %d: because it would exceed the data square width %d", x, y, len(newCol), ds.width)
```

Comment:
Why the codecov still complains? As far as I can say, your tests cover this line 🤔

Reply:
idk, it look like the test does cover this

<img width="1728" alt="Screenshot 2023-06-23 at 2 31 37 PM" src="https://github.com/celestiaorg/rsmt2d/assets/3699047/d0d5af59-f0a1-4cc2-a072-f5d680d461a5">